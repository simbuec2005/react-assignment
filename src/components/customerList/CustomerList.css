.container {
    display: flex;
}

.customet-list {
    flex-basis: 50%;
}

.customet-list-header {
    display: flex;
    border: 1px solid #ccc;
    background-color: #efefef;
}

.customet-list-header .col {
    flex: 1;
    padding: 10px;
}