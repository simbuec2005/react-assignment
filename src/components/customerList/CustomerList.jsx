import React from 'react';
import Customer from '../customer/Customer';
import AddressContainer from '../../containers/AddressContainer';
import './CustomerList.css';

class CustomerList extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            customer: null
        }
    }

    componentDidMount() {
        this.props.getCustomers();
    }

    handleCustomerSelection = (customer) => {
        this.setState({
            customer
        })
    }

    render() {
        const { customers } = this.props;
        const { customer } = this.state;
        return (
            <div className="container">
                <div className="customet-list">
                    <div className="customet-list-header">
                        <div className="col">ID</div>
                        <div className="col">NAME</div>
                        <div className="col">AGE</div>
                    </div>
                    {
                        customers.loading ? <div>Loading...</div> : customers.data.map(customer => {
                            return <Customer key={customer.id} detail={customer} handleClick={this.handleCustomerSelection} />
                        })
                    }
                </div>

                {customer && <AddressContainer customer={customer} />}
            </div>
        )
    }

}

CustomerList.defaultProps = {
    customers: {
        data: [],
        loading: false
    }
}

export default CustomerList;