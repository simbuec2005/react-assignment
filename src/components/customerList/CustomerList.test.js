import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import CustomerList from './CustomerList';

const store = createStore(() => { return {}; });

describe('<CustomerList />', () => {
  test('Should match snapshot', () => {
    const props = {
      customers: {
        data: [],
        loading: false
      },
      getCustomers: () => {}
    }
    const component = create(
      <CustomerList {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
})
