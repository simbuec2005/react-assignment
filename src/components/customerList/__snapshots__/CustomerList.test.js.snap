// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerList /> Should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="customet-list"
  >
    <div
      className="customet-list-header"
    >
      <div
        className="col"
      >
        ID
      </div>
      <div
        className="col"
      >
        NAME
      </div>
      <div
        className="col"
      >
        AGE
      </div>
    </div>
  </div>
</div>
`;
